railties/lib/rails/test_help.rb

```diff
@@ -44,3 +44,7 @@ def before_setup # :nodoc:
     super
   end
 end
+
+if defined?(Rails.application) && Rails.application && !Rails.application.task_loaded?
```

Comment:
Why do we need a `defined?` check? The `before_setup` methods call `Rails.application` without it.

Are Rails engines using this file? They still define an application method though, right?

How about a `load_tasks_once` method:

```ruby
def load_tasks(app = self)
  @load_task_ran  = true
  …
end

def load_tasks_once # :nodoc:
  load_tasks unless defined?(@load_task_ran) && @load_task_ran`
end
```

Though I dislike that the `load_tasks` method has a dynamic app argument — how do we know if we've required the right tasks then?